### PR TITLE
RHDEVDOCS-5631: Adding procedural examples for builds

### PR DESCRIPTION
**Pull request by shipsing**
**Purpose**: To resolve this issue: https://issues.redhat.com/browse/RHDEVDOCS-5631

**Aligned team**: DevTools

**Version for cherrypicking**: build-docs-1.0

**Content for preview**: https://66522--docspreview.netlify.app/openshift-builds/latest/work_with_builds/using-builds

**SME review**:  @apoorvajagtap @jkhelil 

**QE review**:  @jitendar-singh 

**Peer review**:  @adellape

### CODE REVIEW

File: modules/ob-creating-a-buildah-build.adoc [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    "kind": "BuildRun",
+    "name": "buildah-golang-buildrun",
+    "uid": "676dd1da-6ec2-4c04-86b2-cfe5be265260"
+  }
+]
+
+$ oc get pods
+NAME                                READY     STATUS       RESTARTS    AGE
+buildah-golang-buildrun-dtrg2-pod   0/2       Init:0/2     0           9s
+----
```
Comment (apoorvajagtap) [inline modules/ob-creating-a-buildah-build.adoc:100]:
```suggestion
```
We can remove this complete block. This kind of shows up v1alpha1 API, so let's avoid that.. Also, this one's not really mandatory & was just for information.

File: modules/ob-creating-a-buildah-build.adoc [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    value: Dockerfile
+  output: <4>
+    image: image-registry.openshift-image-registry.svc:5000/buildah-example/sample-go-app
+EOF
+----
+<1> The location where the source code is placed.
+<2> The build strategy that you use to build the container.
+<3> A name-value list to specify values for parameters defined in the build strategy. To set the value of the `dockerfile` strategy parameter, specify the Dockerfile location required to build the output image.
+<4> The location where the built image is pushed. By default, the built image is pushed to the {ocp-product-title} cluster internal registry. `buildah-example` is the name of the current project. Ensure that the specified project exists to allow the image push.
+
```
Comment (jkhelil) [inline modules/ob-creating-a-buildah-build.adoc:56]: We need a step here to check the build
 To view the *Build* which you just created:
oc get builds.shipwright.io
NAME                   REGISTERED   REASON      BUILDSTRATEGYKIND      BUILDSTRATEGYNAME   CREATIONTIME
buildah-golang-build   True         Succeeded   ClusterBuildStrategy   buildah             16s

And explanation for the output
@apoorvajagtap can you please give these

File: modules/ob-creating-a-buildah-build.adoc [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  name: buildah-golang-buildrun
+spec:
+  build:
+    name: buildah-golang-build <1>
+EOF
+----
+<1> The `spec.build.name` field denotes the respective build to run, which is expected to be available in the same namespace.
++
+The `BuildRun` resource creates a `TaskRun` resource, which then creates the pods to execute build strategy steps.
+
```
Comment (jkhelil) [inline modules/ob-creating-a-buildah-build.adoc:74]: I can see the verification are done at the end
wondering if we should rather do each verification after commands

so here for thebuildrun we retrive the buildrun
Wait until your *BuildRun* is completed and then you can view it as follows:

oc get buildruns
  
  NAME                              SUCCEEDED   REASON      STARTTIME   COMPLETIONTIME
  buildpack-nodejs-buildrun-xyzds   True        Succeeded   69s         2s

File: modules/ob-creating-a-s2i-build.adoc [hunk 4]
```diff
@@ -0,0 +1,112 @@
+// This module is included in the following assembly:
+//
+// * work-with-builds/using-builds.adoc
+
+:_content-type: PROCEDURE
+[id="ob-creating-a-s2i-build_{context}"]
+= Creating a source-to-image build
+
+You can create a `source-to-image` build and push the created image to a custom quay repository.
```
Comment (adellape) [inline modules/ob-creating-a-s2i-build.adoc:9]:
```suggestion
You can create a `source-to-image` build and push the created image to a custom Quay repository.
```

File: modules/ob-creating-a-s2i-build.adoc [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    value: “quay.io/centos7/nodejs-12-centos7”
+  output: <4>
+    image: quay.io/<repo>/s2i-nodejs-example
+    pushSecret: registry-credential
+EOF
+----
+<1> The location where the source code is placed.
+<2> The build strategy that you use to build the container.
+<3> A name-value list to specify values for parameters defined in the build strategy. To set the value of the `builder-image` strategy parameter, specify the builder image location required to build the output image.
+<4> The location where the built image is pushed. You must push the built image to a custom quay repo. `pushSecret` is the secret name that stores the credentials for pushing container images. To generate a secret of the type `docker-registry` for authentication, see _Authentication to container registries_ in the _Additional resources_ section.
```
Comment (adellape) [inline modules/ob-creating-a-s2i-build.adoc:56]:
```suggestion
<4> The location where the built image is pushed. You must push the built image to a custom Quay repo. `pushSecret` is the secret name that stores the credentials for pushing container images. To generate a secret of the type `docker-registry` for authentication, see _Authentication to container registries_ in the _Additional resources_ section.
```

File: modules/ob-creating-a-buildah-build.adoc [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+  output: <4>
+    image: image-registry.openshift-image-registry.svc:5000/buildah-example/sample-go-app
+EOF
+----
+<1> The location where the source code is placed.
+<2> The build strategy that you use to build the container.
+<3> A name-value list to specify values for parameters defined in the build strategy. To set the value of the `dockerfile` strategy parameter, specify the Dockerfile location required to build the output image.
+<4> The location where the built image is pushed. By default, the built image is pushed to the {ocp-product-title} cluster internal registry. `buildah-example` is the name of the current project. Ensure that the specified project exists to allow the image push.
+
+. Create a `BuildRun` resource and and apply it to the {ocp-product-title} cluster:
```
Comment (adellape) [inline modules/ob-creating-a-buildah-build.adoc:57]:
```suggestion
. Create a `BuildRun` resource and apply it to the {ocp-product-title} cluster:
```

File: modules/ob-creating-a-s2i-build.adoc [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+s2i-nodejs-buildrun-phxxm-pod       0/2     Completed  0          2m
+
+$ oc get tr
+NAME                           SUCCEEDED  REASON     STARTTIME   COMPLETIONTIME
+s2i-nodejs-buildrun-phxxm      True       Succeeded  2m39s        13s
+
+$ oc get br
+NAME                     SUCCEEDED   REASON       STARTTIME     COMPLETIONTIME
+s2i-nodejs-buildrun      True        Succeeded    2m41s           15s
+----
```
Comment (adellape) [inline modules/ob-creating-a-s2i-build.adoc:98]: Same suggestion as above re: breaking up the commands/outputs.

File: modules/ob-creating-a-buildah-build.adoc [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+[NOTE]
+====
+The pod might switch to a `NotReady` state because one of the containers has completed its task. This is an expected behavior.
+====
+
+. Validate whether the image has been pushed to the registry that is specified in the `build.spec.output.image` field. To do so, run the following command from a node that can access the internal registry:
++
+[source,terminal]
+----
+$ podman pull image-registry.openshift-image-registry.svc:5000/buildah-example/sample-go-app
```
[the remark this answers is not in the document]
Comment (shipsing) [reply modules/ob-creating-a-buildah-build.adoc:110]: Thank you for this suggestion :)

**Review by jkhelil — APPROVED** [review]

**Review by apoorvajagtap — CHANGES_REQUESTED** [review]

File: modules/ob-creating-a-buildah-build.adoc [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+<1> The location where the source code is placed.
+<2> The build strategy that you use to build the container.
+<3> A name-value list to specify values for parameters defined in the build strategy. To set the value of the `dockerfile` strategy parameter, specify the Dockerfile location required to build the output image.
+<4> The location where the built image is pushed. By default, the built image is pushed to the {ocp-product-title} cluster internal registry. `buildah-example` is the name of the current project. Ensure that the specified project exists to allow the image push.
+
+. Run the following command to check if the `Build` resource is created:
++
+[source,terminal]
+----
+oc get builds.shipwright.io <build_resource_name>
```
[the remark this answers is not in the document]
Comment (shipsing) [reply modules/ob-creating-a-buildah-build.adoc:61]: Thank you for this suggestion

File: modules/ob-creating-a-s2i-build.adoc [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+  source: <1>
+    git:
+      url: https://github.com/shipwright-io/sample-nodejs
+    contextDir: source-build/
+  strategy: <2>
+    name: source-to-image
+    kind: ClusterBuildStrategy
+  paramValues: <3>
+  - name: builder-image
+    value: “quay.io/centos7/nodejs-12-centos7”
```
Comment (apoorvajagtap) [inline modules/ob-creating-a-s2i-build.adoc:48]:
```suggestion
    value: quay.io/centos7/nodejs-12-centos7
```
The `"` is leading to a failure at the later stages, can we please remove these?

**Review by apoorvajagtap — CHANGES_REQUESTED** [review]

File: work_with_builds/using-builds.adoc [hunk 11]
```diff
@@ -0,0 +1,27 @@
+:_content-type: ASSEMBLY
+[id="using-builds"]
+include::_attributes/common-attributes.adoc[]
+= Using {builds-shortname}
+:context: using-builds
+
+toc::[]
+
+
+After installing {builds-shortname}, you can create a `buildah` or `source-to-image` build for use. You can also delete resources that are not required for a build.
```
Comment (apoorvajagtap) [inline work_with_builds/using-builds.adoc:10]: If we can explicitly specify what resources we are talking to be deleted here, that'll be great.

**Comment by apoorvajagtap** [comment]
LGTM.

**Pull request merged**